### ISSUE TITLE
Boot from SAN - Mezzanine

### ISSUE THREAD

**Issue by 5truja** [HewlettPackard/POSH-HPOneView#107]
Please fill in as much information as possible to help resolve your issue.

### Expected Behavior
Trying to create connection in template (UEFI) having boot from array parameters (primary and secondary and lun 
Connection using boot from SAN array fails.
### Actual Behavior
Creation fails with message below.
This is command line:
$con7 = Get-HPOVNetwork "Fabric-A" -ApplianceConnection test.testdomain.se | New-HPOVServerProfileConnection -connectionType FibreChannel -bootable -priority Primary -arrayWWPn "50:00:14:42:a0:3d:2a:05" -LUN 0 -requestedBW 8000 -name "Fabric-A" -id 7

This is error message:
Name            : Create
Owner           : Administrator
Created         : 2016-04-28T14:04:07.376Z
Duratuon        : 00:00:00
TaskState       : Error
Complete        : 100
ProgressUpdates : {}
TaskErrors      : {@{errorSource=; nestedErrors=System.Object[]; details=; reco
                  mmendedActions=System.Object[]; data=; errorCode=NotSupported
                  AdapterBIOS; message=The 'Use Adapter BIOS' Fibre Channel boo
                  t option cannot be selected with UEFI Boot Mode.}}
### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion`):
PowerShell Library Version : 2.0.202.0
HPE OneView Appliance Version:  
Output from `$PSVersionTable` on your Windows Host:

**Comment by ChrisLynchHPE** [comment]
You will need to provide a bit more information for me.  You are using an old library version.  Please update to the 2.0.404.0 release.  In it are two new Cmdlets for debugging purposes: `Start-HPOVLibraryTrace` and `Stop-HPOVLibraryTrace`.  Do know that they do not support the PowerShell ISE as it is a different 'host' environment that requires a different method of capturing command output.  Provide the trace file that is created here so I can look at the Server Profile object you are trying to create.

**Comment by 5truja** [comment]
> Do know that they do not support the PowerShell ISE as it is a different 'host' environment that requires a different method of capturing command output
Like what environemt? Would you please give som example.

**Comment by ChrisLynchHPE** [comment]
There are two environments you can run PowerShell scripts from:


·       PowerShell ISE

·       PowerShell Command Prompt

The Start-HPOVLibraryTrace makes a call to Start-Transcript, which can only be run from a PowerShell Command Prompt, not the PowerShell ISE.  I document this in the Cmdlet help.

**Comment by 5truja** [comment]
Hi,
Here is the trace.

Kind regards

**Comment by 5truja** [comment]
Hi Chris,

Please correct me if I m wrong but when taking the same script with manually managed Boot from SAN parameters it works but when using -bootable -priority Primary –arrayWWPn switches if fails.

According to error message “Use Adapter BIOS” seems to be chosen by default for some reason: 'Use Adapter BIOS' Fibre Channel boot option cannot be selected with UEFI Boot Mode.}}

Anyway when I change to GUI and change it to “Specify boot targets” it works. How to get this “by cmdlet” cmd line instead?

[cid:image001.png@01D1A170.CA255B70]

From: Brekalo Mladen
Sent: den 28 april 2016 16:59
To: 'HewlettPackard/POSH-HPOneView' <reply@reply.github.com>; HewlettPackard/POSH-HPOneView <POSH-HPOneView@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [HewlettPackard/POSH-HPOneView] Boot from SAN - Mezzanine (#107)

Hi,
Here is the trace.

Kind regards

**Comment by ChrisLynchHPE** [comment]
You didn't attach any file to your reply.

Specifying the BootPriority to Primary and providing a Boot Target should not set the connection to Use BIOS.  I'll know more when you provide the trace file that was generated.

**Comment by 5truja** [comment]
Wrong format. Here it is.
[1604281655_HPOV_transcript.zip](https://github.com/HewlettPackard/POSH-HPOneView/files/240979/1604281655_HPOV_transcript.zip)

**Comment by ChrisLynchHPE** [comment]
This looks to be an API bug in the error message that is returned.  In looking at the server profile object you are sending, I see Connection 7 & 8 are both set to `Primary` priority:

``` JSON
{
	"id" : 7,
	"functionType" : "FibreChannel",
	"name" : "SHL-V-Fabric-A",
	"portId" : "Auto",
	"networkUri" : "/rest/fc-networks/84dd6f77-fb98-44f7-98af-8e351f4b11c5",
	"requestedMbps" : 8000,
	"boot" :
	{
		"priority" : "Primary",
		"targets" : [
			{
				"arrayWwpn" : "50:00:14:42:a0:3d:2a:01",
				"lun" : "0"
			}
		]
	}
},
{
	"id" : 8,
	"functionType" : "FibreChannel",
	"name" : "SHL-V-Fabric-B",
	"portId" : "Auto",
	"networkUri" : "/rest/fc-networks/0954dae4-2469-4333-aa5a-16bb4c78162c",
	"requestedMbps" : 8000,
	"boot" :
	{
		"priority" : "Primary",
		"targets" : [
			{
				"arrayWwpn" : "50:00:14:42:b0:3d:5d:00",
				"lun" : "0"
			}
		]
	}
}```

You must change one of them to be Secondary.  I will look at adding logic to the `New-HPOVServerProfile` and `New-HPOVServerProfileTemplate` Cmdlets to help check for this condition.

**Comment by 5truja** [comment]
Thanks.
I will check that.

Kind regards

**Comment by 5truja** [comment]
I tested few times and yes there was primary twice (but it was not there the first time). 
[1604282151_HPOV_transcript.zip](https://github.com/HewlettPackard/POSH-HPOneView/files/241338/1604282151_HPOV_transcript.zip)
Now I changed one of them to Secondary but still the same error message. Here is new trace.

**Comment by ChrisLynchHPE** [comment]
This is very strange.  I still think this is an Appliance bug, as the correct object properties are being sent to the appliance.  I'm researching internally.

In the meantime, can you do the following via your Web Browser (IE11 is just fine):

1. Navigate to the appliance, and go to the Create Server Profile Template editor.
2. Fill in all of the values you need.
3. Press F12 to bring up Developer Tools.
4. Select the Network Tab.  Make sure that the `Always refresh from server` button is UNSELECTED, and click the Green `Play` button.
5. Go back to your browser and click the Create button.
6. In the F12 Developer Tools screen, you will see a whole bunch of entries.  Look for a row with the `POST` method and `202` result.  Double-click on it (which will take you to the Advanced View of that entry), then select the `Request Body` tab.  Provide the contents of that here.

**Comment by 5truja** [comment]
1.  > Make sure that the Always refresh from server button is UNSELECTED, and click the Green Play button.
Strange but I really do not see any “Always refresh from server button”. I don’t se play button either
I do see other buttons like “Record network log”, “Capture screenshot”  etc

**Comment by 5truja** [comment]
> Web Browser (IE11 is just fine):
Sorry now I see this let me try again…(I use chrome)

**Comment by ChrisLynchHPE** [comment]
> I use chrome
No one is perfect! ;-)

Also, I didn't see that you were trying to create a Server Profile Template.  It looks like the Connections behavior is a bit different, but has never come up before in my testing.  I will need to look into this further.

**Comment by 5truja** [comment]
Here is the the Post 200 request
[createServerTemplatePost_200.zip](https://github.com/HewlettPackard/POSH-HPOneView/files/241452/createServerTemplatePost_200.zip)

**Comment by ChrisLynchHPE** [comment]
Thanks.  I found the problem, and will have it fixed in an upcoming release.

**Comment by 5truja** [comment]
What is the problem?

**Comment by ChrisLynchHPE** [comment]
There is a property that is required, which is missing in the object.  So when the object is then sent to the API, it sets the Priority to ‘UseAdapterBios’.  In turn, that causes the async task failure.  This is only specific to Server Profile Templates.

**Comment by 5truja** [comment]
Thanks. On the other side I actually have issue when applying server profile that boots from san. Applying fails (cannot find targets on array). It is GEN9. I have 2 paths in server profile that could be configured and storage guys are usualy configuring 2 additional. GEN 8 has no these kinds of issues. I am not sure if this is GEN9 or OV.

Med v?nlig h?lsning,
Mladen Brekalo


28 apr. 2016 kl. 23:36 skrev Chris Lynch <notifications@github.com<mailto:notifications@github.com>>:

There is a property that is required, which is missing in the object. So when the object is then sent to the API, it sets the Priority to 'UseAdapterBios'. In turn, that causes the async task failure. This is only specific to Server Profile Templates.

**Comment by ChrisLynchHPE** [comment]
That is a condition of UEFI and specifying Boot Targets.  If the System ROM cannot find a boot device, then the Server Profile fails to apply.  That particular condition is not something I can help address or fix with the Library.  If you change the Boot Mode to Legacy BIOS, you can avoid that from happening.  You still get BIOS Setting monitoring in Gen9.  You just lose UEFI access from the OS.

**Comment by 5truja** [comment]
Thanks. I tried Legacy BIOS many times but it does not find the disk. I've got it working (boot from san) using uefi and single path. In that case i need to check with storage guys. Legay bios did work with gen8, but as you said it is "out of scooe".

Kind regards


28 apr. 2016 kl. 23:53 skrev Chris Lynch <notifications@github.com<mailto:notifications@github.com>>:

That is a condition of UEFI and specifying Boot Targets. If the System ROM cannot find a boot device, then the Server Profile fails to apply. That particular condition is not something I can help address or fix with the Library. If you change the Boot Mode to Legacy BIOS, you can avoid that from happening. You still get BIOS Setting monitoring in Gen9. You just lose UEFI access from the OS.

**Comment by 5truja** [comment]
Hi Chris,

After checking zoning I’ve installed two GEN9 and it works fine BUT there is a strange behaviour with templates.
I have template with boot from SAN parameters (-priority –arrayWWPn and LUN ID). Everything worked fine until I applied server profile. Server profile inherits boot from SAN parameters but the change to manual on both FibreChannel connections in template.
This means I cannot use the template next time to create new server profile having Boot from san parameters (with priority and –arraWWPn). I can only copy from the very first generated server profile in order to inherit these parameters.

I’ve seen this behavior before and I have even opened the case within HPE. I’ve got no answer after two weeks. Still waiting for an explanation.

Did not want to bother you with this but I hope you know anything about that.
I can create separate case (HPE case id #4771479352) but it was easier to “chain” to this discussion since it is kind of connected because when scripting priority fails it must be more than that (of course I am speculating).

Kind regards

**Comment by ChrisLynchHPE** [comment]
> Server profile inherits boot from SAN parameters but the change to manual on both FibreChannel connections in template.

Sorry, but I cannot reproduce that behavior in my lab.  I'd suggest you open a support case.

**Comment by 5truja** [comment]
Yes. I did it already two weeks ago. Thanks

Med v?nlig h?lsning,
Mladen Brekalo


29 apr. 2016 kl. 16:43 skrev Chris Lynch <notifications@github.com<mailto:notifications@github.com>>:


Server profile inherits boot from SAN parameters but the change to manual on both FibreChannel connections in template.

Sorry, but I cannot reproduce that behavior in my lab. I'd suggest you open a support case.

-

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 2.0.414.0](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v2.0.414.0).

Closing.

**Comment by 5truja** [comment]
Thanks

Med v?nlig h?lsning,
Mladen Brekalo


5 maj 2016 kl. 05:10 skrev Chris Lynch <notifications@github.com<mailto:notifications@github.com>>:


This is fixed in Release 2.0.414.0<https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v2.0.414.0>.

Closing.

-
